src/hooks/login/useWalletConnectV2Login.ts

```diff
@@ -0,0 +1,243 @@
+import { useRef, useState } from 'react';
+import {
+  WalletConnectProviderV2,
+  PairingTypes
+} from '@elrondnetwork/erdjs-wallet-connect-provider';
+import { useUpdateEffect } from 'hooks/useUpdateEffect';
+import {
+  getAccountProvider,
+  setAccountProvider
+} from 'providers/accountProvider';
+import { loginAction } from 'reduxStore/commonActions';
+import { useDispatch, useSelector } from 'reduxStore/DappProviderContext';
+import { isLoggedInSelector } from 'reduxStore/selectors/loginInfoSelectors';
```

Comment:
This file looks very similar to useWalletConnectLogin for v1, can we extract the code that is used by both hooks into a general hooks and use that one, instead of copy pasting the logic?

Reply:
We will keep the hooks separate for now since WC2 is still in a beta phase and more breaking changes might apprear. This way any changes won't affect WC1 that is already in a stable release.